chart/templates/_helpers.yaml

```diff
@@ -224,27 +224,30 @@ If release name contains chart name it will be used as a full name.
       value: "false"
     {{- end }}
     {{ else if .Values.dags.gitSync.credentialsSecret }}
+    {{- if semverCompare "<v4.0.0" .Values.images.gitSync.tag }}
```

Comment:
We can't rely on the tag alone to determine the version, as some folks could be mirroing these images (or building compatible ones) with a different tagging strategy.

If we need to branch based on it, we will need a new version field. I think it's easier to just set both v3 and v4 env vars concurrently, like we have (or intended to I guess!) for everything.

Reply:
Could we require one of the pairs, and automate the other?
```yaml
env:
  - name: GITSYNC_USERNAME
    value: $(GIT_SYNC_USERNAME)
  - name: GITSYNC_PASSWORD
    value: $(GIT_SYNC_PASSWORD)
```

Reply:
Or set them all to `optional: true` and let runtime errors get raised.

Reply:
Okay, I see what's going on.

> Could we require one of the pairs, and automate the other?

That would be relatively easy, yeah. We'd just pick, probably the v3 form since that's been around and working, and expect the key to be that in the sercet.

> Or set them all to `optional: true`

I'm good that with too.

I think originally it was intended for folks to set both the v3 and v4 forms in the secret, but that's with the benefit of hindsight and I would have also just done the v4 first...

I think the v3 key in the secret only approach is a bit cleaner, so I'd probably lean that way?